### DISCUSSION TITLE
VS Code multi-root workspace cross-references

### DISCUSSION THREAD

**Discussion by freenandes** [marp-team/marp#487] [Q&A] [answered]
Is it possible for the extension to fetch the theme from a root folder that is not the open folder (with the *.md)?

I'm trying all possible combinations and looking to prevent typos, but i cannot get the relative path to the theme on another root folder.
```
workspace
 │  
 ├—rootFolderA
 │  └—content.md
 │  
 └—rootFolderB
    └—theme.css
```

I tried
- the recommended, i think `${workspaceFolder:rootFolderB}/theme.css`
- `rootFolderB/theme.css`
- `../rootFolderB/theme.css`
- `../../rootFolderB/theme.css`
- ...

Is it not supported? Any strategy to achieve this?

**Comment by yhatt** [comment]
No, that is not supported.

The path resolution rule for custom theme settings in Marp for VS Code follows the configuration of the standard Markdown preview in VS Code: `markdown.styles`. Due to potential security issues, both settings do not allow using the style across different workspace roots.

> * https://github.com/microsoft/vscode/issues/45260#issuecomment-371428889
> * https://github.com/microsoft/vscode/issues/77290#issuecomment-515729785
> * https://github.com/microsoft/vscode/issues/96201
> * https://github.com/microsoft/vscode/issues/99280
> * https://github.com/microsoft/vscode/issues/101991
>
> _&mdash; https://github.com/orgs/marp-team/discussions/115#discussioncomment-7435455_

**Reply by freenandes** [reply]
oh... Thanks anyway 🙌

**Comment by freenandes** [comment, marked as answer]
Before, I had created a workspace made from folderA and then added folderB to it, making both folders workspace roots in a concept in VS Code called multi-root workspace. This won't work with Marp, as pointed in the replies.

I actually had to create a new workspace folder, and then move both folders into it, each with its git.

```
workspace
│  
└—workspaceRoot
  │ 
  ├—folderA
  │ │ 
  │ └—content.md
  │ 
  └—folderB
    │ 
    └—theme.css
```
So the reference would simply be `folderB/theme.css` for this workspace.

This is fine and it works on most extensions. Git works fine.
